### DISCUSSION TITLE
Contributions

### DISCUSSION THREAD

**Discussion by jgarte** [Abjad/abjad#1466] [Q&A] [answered]
@trevorbaca Hi, what is the notion of [contribution](https://github.com/Abjad/abjad/blob/ebf0f91966bea1c1e55f1160a63c68aa6dfc84aa/abjad/contributions.py) generally? Or, what is some background on the reason for choosing that name. I'm just trying to understand that I'm thinking about contributions correctly in this context.

**Comment by trevorbaca** [comment, marked as answer]
Hi @jgarte, a (formatting) contribution is any string (or strings) that either a component (note, rest, chord, tuplet, voice, etc) or an indicator (articulation, clef, dynamic, and essentially everything else) "contributes" to the LilyPond output produced when we call `abjad.lilypond()`.

The (lexical) place to which a formatting contribution is written is the "site" of that contribution in the LilyPond output.
